repo_policy_compliance/__init__.py

```diff
@@ -59,6 +60,60 @@ class Report(NamedTuple):
     reason: str | None
 
 
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def log_check(func: Callable[P, R]) -> Callable[P, R]:
+    """Log before check and result of check.
+
+    Args:
+        func: The function that executes a check.
+
+    Returns:
+        The function where the check is logged before it starts and the results are logged.
+    """
+    check_name = func.__name__
+
+    def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
+        """Replace function.
+
+        Args:
+            args: The positional arguments passed to the method
+            kwargs: The keywords arguments passed to the method
+
+        Returns:
+            The return value after calling the wrapped function.
+        """
+        logging.info("start check '%s'", check_name)
+        result = func(*args, **kwargs)
+        logging.info("check '%s' finished, result: %s", check_name, result)
+        return result
+
+    return wrapper
+
+
+def _setup_logging() -> None:
+    """Initialise logging for check execution."""
+    handler = logging.FileHandler(filename="/dev/stdout")
+    handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter("%(asctime)s - %(message)s")
+    handler.setFormatter(formatter)
+
+    # Setup local logging
+    logger = logging.getLogger()
+    logger.setLevel(logging.DEBUG)
+    logger.addHandler(handler)
+
+    # Setup urllib3 logging
+    urllib3_logger = logging.getLogger("urllib3")
+    urllib3_logger.setLevel(logging.DEBUG)
+    urllib3_logger.addHandler(handler)
```

Comment:
Are we sure we want to set debug level log ?

Reply:
Yes for now because we had an issue where the GitHub API requests timed out and need to see what happened there. We can reduce the level if we don't encounter the issue again/ we encounter it and resolve it. It will also help us understand how long a given request takes and the number of API calls involved

Reply:
As discussed, this is for us to gather data in production regarding timeout issues